services/media/service/src/domains/collections/localization/collections-images-replication-handlers.ts

```diff
@@ -0,0 +1,101 @@
+import { OwnerPgPool } from '@axinom/mosaic-db-common';
+import { Dict } from '@axinom/mosaic-service-common';
+import { selectOne } from 'zapatos/db';
+import { collections_images } from 'zapatos/schema';
+import { Config } from '../../../common';
+import {
+  assertLocalizationColumn,
+  assertLocalizationData,
+  getUpsertMessageData,
+  ReplicationOperationHandlers,
+} from '../../common';
+import { LOCALIZATION_COLLECTION_TYPE } from './constants';
+
+type LocalizableCollectionImage = Pick<
+  collections_images.JSONSelectable,
+  'collection_id' | 'image_id' | 'image_type'
+> & { [k: string]: unknown };
+
+const assertCollectionImage: (
+  data: Dict<unknown> | undefined,
+) => asserts data is LocalizableCollectionImage = (
+  data: Dict<unknown> | undefined,
+): asserts data is LocalizableCollectionImage => {
+  assertLocalizationData(data, 'collections_images');
+  assertLocalizationColumn(data, 'collection_id', 'collections_images');
+  assertLocalizationColumn(data, 'image_id', 'collections_images');
+  assertLocalizationColumn(data, 'image_type', 'collections_images');
+};
+
+const collectionIsDeleted = async (
+  collectionImage: LocalizableCollectionImage,
+  ownerPool: OwnerPgPool,
+): Promise<boolean> => {
+  const data = await selectOne(
+    'collections',
+    { id: collectionImage.collection_id },
+    { columns: ['id'] },
+  ).run(ownerPool);
+  return !data;
+};
+
+export const collectionsImagesReplicationHandlers = (
+  config: Config,
+  ownerPool: OwnerPgPool,
+): ReplicationOperationHandlers => {
+  const entityType = LOCALIZATION_COLLECTION_TYPE;
+  return {
+    insertHandler: async (newData: Dict<unknown> | undefined) => {
```

Comment:
We already have `getUpsertMessageData`, so I would say creating an `upsertHandler` would not improve it too much and might actually lose some readability. I think I'd like to keep it as is for now, it would also make it easier to add possible customizations in the future.